### PR TITLE
Add support for underline navigation styling

### LINKED ISSUE #784: Add `navset_underline()`, `navset_card_underline()`, and `page_navbar(underline = T/F)`
`.nav-underline` was added in Bootstrap 5.3 and is convenient for achieving the look for we're aiming for in `bs_theme(preset="shiny")` 

https://getbootstrap.com/docs/5.3/components/navs-tabs/#underline

### PR DESCRIPTION
**Pull request by cpsievert**
Closes #784

### CODE REVIEW

File: man/figures/page-navbar.png
Comment (gadenbuie) [inline man/figures/page-navbar.png:1]: It seems like there's some padding below the underline that I found surprising. I was expecting that the border be aligned with the bottom edge of the navbar area.

Here's a stackblitz from an example in the Bootstrap docs: https://stackblitz.com/edit/dyaf7f?file=index.html

Comment (gadenbuie) [reply man/figures/page-navbar.png:1]: Oh wait, no that's expected: https://stackblitz.com/edit/fhmwnq?file=index.html

**Review by gadenbuie — APPROVED** [review]
This looks good to me! I like the abstraction of `navset_card()`.

It'd be nice to include `navset_underline()` in the navset examples here: https://github.com/rstudio/bslib/blob/nav-underline/man/fragments/ex-navset_tab.Rmd

File: man/figures/page-navbar.png
Comment (cpsievert) [reply man/figures/page-navbar.png:1]: Yea, nice catch. I suppose if we feel like the "bottom alignment is important", we can look into tweaks for `bs_theme(preset="shiny")`

**Pull request merged**